File: tests/expectations/rename_crate.java

import com.sun.jna.*;
import com.sun.jna.ptr.*;

enum BindingsSingleton {
  INSTANCE;
  final Bindings lib = Native.load("rename_crate", Bindings.class);
}

interface Bindings extends Library {
  Bindings INSTANCE = BindingsSingleton.INSTANCE.lib;
  class ContainsNoExternTy {}



  @Structure.FieldOrder({"x"})
  class Foo extends Structure implements Structure.ByValue {
    public Foo() {
      super();
    }

    public Foo(Pointer p) {
      super(p);
    }

    public int x;

  }

  @Structure.FieldOrder({"x"})
  class FooByReference extends Structure implements Structure.ByReference {
    public FooByReference() {
      super();
    }

    public FooByReference(Pointer p) {
      super(p);
    }

    public int x;

  }



  @Structure.FieldOrder({"y"})
  class RenamedTy extends Structure implements Structure.ByValue {
    public RenamedTy() {
      super();
    }

    public RenamedTy(Pointer p) {
      super(p);
    }

    public long y;

  }

  @Structure.FieldOrder({"y"})
  class RenamedTyByReference extends Structure implements Structure.ByReference {
    public RenamedTyByReference() {
      super();
    }

    public RenamedTyByReference(Pointer p) {
      super(p);
    }

    public long y;

  }


  void root(Foo a);

  void renamed_func(RenamedTy a);

  void no_extern_func(ContainsNoExternTy a);

}

File: tests/expectations/rename_crate.zig
const std = @import("std");

pub const Foo = extern struct {
    x: i32,
};
pub const RenamedTy = extern struct {
    y: u64,
};
pub const NoExternTy = extern struct {
    field: u8,
};
pub const ContainsNoExternTy = extern struct {
    field: NoExternTy,
};

pub extern fn root(a: Foo) anyopaque;

pub extern fn renamed_func(a: RenamedTy) anyopaque;

pub extern fn no_extern_func(a: ContainsNoExternTy) anyopaque;
